source/is-literal.d.ts

```diff
@@ -3,6 +3,22 @@ import type {Numeric} from './numeric';
 import type {IsNotFalse} from './internal';
 import type {IsNever} from './is-never';
 
+/**
+Returns a boolean for whether the given type `T` is Primitive Value or Primitive Type.
+@example
+```
+IsPrimitive<'string'>
+//=> true
+
+IsPrimitive<string>
+//=> true
+
+IsPrimitive<Object>
+//=> false
+```
+ */
+type IsPrimitive<T> = [T] extends [Primitive] ? true : false;
```

Comment:
Can you move this to internal.d.ts? It makes it easier to reuse types and easier to see what should be exported publicly later on.